podium-gateway/src/main/java/nl/thehyve/podium/service/RequestService.java

```diff
@@ -669,6 +721,40 @@ public RequestRepresentation requestRevision(
     }
 
     /**
+     * Provide the review feedback for a review round in a request.
+     *
+     * @param user the currently logged in user
+     * @param feedbackRepresentation the feedback supplied by the reviewer
+     *
+     * @return ReviewFeedback the updated review feedback
+     * @throws AccessDenied if the currently logged in user is not the owner of the feedback given
+     */
+    @Timed
+    public ReviewFeedback provideReviewFeedback(
+        AuthenticatedUser user,
+        ReviewFeedbackRepresentation feedbackRepresentation
+    ) {
+        log.debug("Providing review feedback for {}", feedbackRepresentation.getUuid());
+        if (user.getUuid() != feedbackRepresentation.getReviewer().getUuid()) {
```

Comment:
Please move this check to below, where the `ReviewFeedback` object has been fetched from the database. Now you trust the reviewer for information about the ownership.